components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/AuthenticatorManagementDAO.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.dao;
+
+import org.wso2.carbon.identity.application.common.exception.AuthenticatorMgtException;
+import org.wso2.carbon.identity.application.common.model.LocalAuthenticatorConfig;
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.AuthenticationType;
+
+import java.util.List;
+
+/**
+ * This interface performs CRUD operations for the Local Application Authenticator configurations.
+ */
+public interface AuthenticatorManagementDAO {
+
+    /**
+     * Create a new user defined Local Application Authenticator configuration.
+     *
+     * @param authenticatorConfig   Local Application Authenticator configuration.
+     * @param tenantId              Tenant Id.
+     *
+     * @return Created LocalAuthenticatorConfig.
+     * @throws AuthenticatorMgtException If an error occurs while adding the authenticator configuration.
+     */
+    LocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
+            LocalAuthenticatorConfig authenticatorConfig, Integer tenantId, AuthenticationType type)
+            throws AuthenticatorMgtException;
```

Comment:
Shall we move tenantId as the last parameter. Update methods below as well

Reply:
Addressed with commit d72ebd31b7150e7c562faf83ee80300a60e1e5d8